File: app/src/main/java/net/jkcat/jkit/java/JNetApi.java
package net.jkcat.jkit.java;

import net.jkcat.network.BaseResult;
import net.jkcat.network.base.NetworkApi;
import net.jkcat.network.handler.ExceptionHandler;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import io.reactivex.functions.Function;
import okhttp3.Interceptor;

/**
 * @author JokerCats on 2021.05.08
 */
public class JNetApi extends NetworkApi {

    private static volatile JNetApi sInstance;

    public static JNetApi getInstance() {
        if (sInstance == null) {
            synchronized (JNetApi.class) {
                sInstance = new JNetApi();
            }
        }
        return sInstance;
    }

    protected JNetApi() {
        super("http://app.rv888.vip/");
    }

    @Nullable
    @Override
    protected Interceptor getInterceptor() {
        return null;
    }

    @NotNull
    @Override
    protected <T> Function<T, T> getAppErrorHandler() {
        return response -> {
            if (response instanceof BaseResult) {
                BaseResult result = (BaseResult) response;
                if (result.getCode() != 0) {
                    ExceptionHandler.ServerException exception =
                            new ExceptionHandler.ServerException(result.getMsg());
                    exception.setCode(result.getCode());
                    throw exception;
                }
            }
            return response;
        };
    }

}

File: app/src/main/java/net/jkcat/jkit/kotlin/JNetApi.kt
package net.jkcat.jkit.kotlin

import io.reactivex.functions.Function
import net.jkcat.network.base.NetworkApi
import net.jkcat.network.handler.HandlerFunction
import okhttp3.Interceptor

/**
 * @author JokerCats on 2021.05.08
 */
object JNetApi : NetworkApi("http://106.15.33.253:6385/hz/app/") {

    override fun getInterceptor(): Interceptor? {
        return null
    }

    override fun <T> getAppErrorHandler(): Function<T, T> {
        return HandlerFunction()
    }

}